order/command-side/src/test/java/org/example/event/sourcing/order/poc/command/order/producer/OrderEventProducerTest.java

```diff
@@ -0,0 +1,118 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.LoggerFactory;
+import org.springframework.kafka.core.KafkaOperations.OperationsCallback;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.support.SendResult;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.LoggerContext;
+
+class OrderEventProducerTest {
+
+	private static InMemoryAppender inMemoryAppender;
+
+	@SuppressWarnings("unchecked")
+	private KafkaTemplate<String, OrderEvent> kafkaTemplate = mock(KafkaTemplate.class);
+
+	private OrderEventProducer orderEventProducer = new OrderEventProducer(kafkaTemplate);
+
+	private String orderId = "id0";
+	private String topic = "topicX";
+	private int partition = 1;
+	private long timestamp = System.currentTimeMillis();
+
+	private OrderEvent order;
+	private SendResult<String, OrderEvent> sendResult;
+
+	@BeforeAll
+	static void setupLogger() {
+		inMemoryAppender = new InMemoryAppender();
+		inMemoryAppender.setContext((LoggerContext) LoggerFactory.getILoggerFactory());
+		((Logger) LoggerFactory.getLogger(OrderEventProducer.class)).addAppender(inMemoryAppender);
+		inMemoryAppender.start();
+	}
+
+	@BeforeEach
+	void setup() {
+		order = new OrderEvent(orderId, CREATED, Instant.now(), Instant.now());
+		sendResult = buildSendResult();
+		given(kafkaTemplate.executeInTransaction(any())).willAnswer(invocation -> {
+			@SuppressWarnings("unchecked")
+			OperationsCallback<String, OrderEvent, Boolean> callback = (OperationsCallback<String, OrderEvent, Boolean>) invocation
+					.getArgument(0, OperationsCallback.class);
+			callback.doInOperations(kafkaTemplate);
+			return true;
+		});
+		inMemoryAppender.reset();
+	}
+
+	@AfterAll
+	static void tearDown() {
+		inMemoryAppender.stop();
+		((Logger) LoggerFactory.getLogger(OrderEventProducer.class)).detachAppender(inMemoryAppender);
+
+	}
+
+	private SendResult<String, OrderEvent> buildSendResult() {
+		ProducerRecord<String, OrderEvent> producerRecord = new ProducerRecord<>(topic, "key", order);
+		RecordMetadata recordMetadata = new RecordMetadata(new TopicPartition(topic, partition), 0, 0, timestamp, 0, 0);
+		return new SendResult<>(producerRecord, recordMetadata);
+	}
+
+	@Test
+	void shouldSend() {
+		given(kafkaTemplate.send(ORDER_TOPIC, orderId, order))
+				.willReturn(CompletableFuture.completedFuture(sendResult));
+
+		orderEventProducer.create(order);
+
+		// verify that kafkaTemplate.send is called
+		then(kafkaTemplate).should().send(ORDER_TOPIC, orderId, order);
+	}
+
+	@Test
+	void shouldLogResult() {
+		given(kafkaTemplate.send(ORDER_TOPIC, orderId, order))
+				.willReturn(CompletableFuture.completedFuture(sendResult));
+
+		orderEventProducer.create(order);
+
+		// verify that sendResult is logged
+		BDDAssertions.then(inMemoryAppender.contains("topic-partition " + topic + "-" + partition, Level.INFO))
```

Comment:
I think we can just use the ListAppender
and use assertion or BDDAssertion to check the log list
```
  // I'm not sure how to transfer the following code to BDDAssertion syntax
  assertThat(listAppender.list)
                 .hasSize(x);
                 // or
                 .extracting(ILoggingEvent::getLevel)
                 .doesNotContains(Lavel.WARN);
                 // or
                 .extracting(ILoggingEvent::getMessage)
                 .containsExactly("some massage with arg= {}");
```

Reply:
Ok, I'll try this later